Fix multiple warnings found by -Wunused-parameter

### Preliminary checks

- [x] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [x] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Android

### Describe the bug

These are the warnings:
```
 C/C++: /home/runner/work/fheroes2/fheroes2/android/app/jni/main/../../../../src/fheroes2/ai/ai_battle_spell.cpp:904:106: warning: unused parameter 'spell' [-Wunused-parameter]
C/C++:   904 | AI::SpellcastOutcome AI::BattlePlanner::spellEarthquakeValue( const Battle::Arena & arena, const Spell & spell, const Battle::Units & friendly ) const
C/C++:       |                                                                                                          ^
C/C++: 1 warning generated.
C/C++: /home/runner/work/fheroes2/fheroes2/android/app/jni/main/../../../../src/fheroes2/maps/map_random_generator_info.cpp:55:59: warning: unused parameter 'record' [-Wunused-parameter]
C/C++:    55 |     void MapStateManager::commitTransaction( const size_t record )
C/C++:       |                                                           ^
C/C++: 1 warning generated.
```

### Save file

No need

### Additional info

It would be great to enable **-Wunused-parameter** flag at least on Android build which caught these places.

--- a/src/fheroes2/ai/ai_battle_spell.cpp
+++ b/src/fheroes2/ai/ai_battle_spell.cpp
@@ -1,6 +1,6 @@
 /***************************************************************************
  *   fheroes2: https://github.com/ihhub/fheroes2                           *
- *   Copyright (C) 2024 - 2025                                             *
+ *   Copyright (C) 2024 - 2026                                             *
  *                                                                         *
  *   This program is free software; you can redistribute it and/or modify  *
  *   it under the terms of the GNU General Public License as published by  *

--- a/src/fheroes2/maps/map_random_generator_info.cpp
+++ b/src/fheroes2/maps/map_random_generator_info.cpp
@@ -1,6 +1,6 @@
 /***************************************************************************
  *   fheroes2: https://github.com/ihhub/fheroes2                           *
- *   Copyright (C) 2025                                                    *
+ *   Copyright (C) 2025 - 2026                                             *
  *                                                                         *
  *   This program is free software; you can redistribute it and/or modify  *
  *   it under the terms of the GNU General Public License as published by  *

